### PR TITLE
Simplify node babel and move cache-dir

### PR DESCRIPTION
**Pull request by jquense**
Use node current in a non production build, and don’t transpile the browser code at all (web pack will do that later)

Also moves the cache dir up a level so as not to be confused with the other src code

### CODE REVIEW

**Comment by KyleAMathews** [comment]
Deploy preview failed.

Built with commit 859001d217b3ef8978d39dd3577bfc56425453e7

https://app.netlify.com/sites/using-glamor/deploys/598c8c597960b12b1ff4bf05

File: packages/gatsby-link/__tests__/index.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
+"use strict";
```
Comment (KyleAMathews) [inline packages/gatsby-link/__tests__/index.js]: This needs ignored

File: .babelrc.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+
```
Comment (KyleAMathews) [inline .babelrc.js]: Lovely!

File: packages/gatsby-plugin-coffeescript/__tests__/gatsby-node.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
+"use strict";
```
Comment (KyleAMathews) [inline packages/gatsby-plugin-coffeescript/__tests__/gatsby-node.js]: plus this and a bunch of others

File: packages/gatsby-link/__tests__/index.js (hunk 1 above)
Comment (jquense) [inline packages/gatsby-link/__tests__/index.js]: arg these. I have this fixed in the webpack PR let me cherry-pick those changes

**Comment by KyleAMathews** [comment]
@jquense this ready to go?

**Comment by jquense** [comment]
I believe so!

File: packages/gatsby-remark-katex/index.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+"use strict";
```
Comment (KyleAMathews) [inline packages/gatsby-remark-katex/index.js]: This needs ignored

Comment (KyleAMathews) [inline packages/gatsby-remark-katex/index.js]: Similar to the other remark plugins

Comment (KyleAMathews) [inline packages/gatsby-remark-katex/index.js]: Fixed this

**Comment by KyleAMathews** [comment]
Tweaked a few more things—will test a bit more later and merge + publish

**Comment by KyleAMathews** [comment]
Thanks @jquense!

**Pull request merged**